src/mmrelay/meshtastic_utils.py

```
@@ -533,15 +533,54 @@ def _get_device_metadata(client: Any) -> dict[str, Any]:
+            future.result(timeout=30.0)
+        except FuturesTimeoutError:
+            timed_out = True
+            logger.debug("getMetadata() timed out after 30 seconds")
+            # If the worker is still running, restore stdio immediately so the
+            # main process does not keep writing to the captured buffer.
+            if redirect_active.is_set():
+                if sys.stdout is output_capture:
+                    sys.stdout = orig_stdout
+                if sys.stderr is output_capture:
+                    sys.stderr = orig_stderr
+        except Exception as e:
```

Ruff (0.14.11)

Do not catch blind exception: `Exception`

(BLE001)